internal/controlplane/handlers_repositories.go

```diff
@@ -491,28 +455,54 @@ func (s *Server) ListRemoteRepositoriesFromProvider(
 	return out, nil
 }
 
-func (s *Server) deleteRepositoryAndWebhook(
+// TODO: this probably can probably be used elsewhere
+// returns project ID and github client - this flow of code exists in multiple
+// places
+func (s *Server) getProjectIDAndClient(
 	ctx context.Context,
-	repo db.Repository,
-	provider db.Provider,
-) error {
-	tx, err := s.store.BeginTransaction()
+	request HasProtoContext,
+) (uuid.UUID, v1.GitHub, error) {
+	projectID := getProjectID(ctx)
+
+	provider, err := getProviderFromRequestOrDefault(ctx, s.store, request, projectID)
+	if err != nil {
+		return uuid.Nil, nil, providerError(err)
+	}
+
+	pbOpts := []providers.ProviderBuilderOption{
+		providers.WithProviderMetrics(s.provMt),
+		providers.WithRestClientCache(s.restClientCache),
+	}
+
+	p, err := providers.GetProviderBuilder(ctx, provider, s.store, s.cryptoEngine, pbOpts...)
```

Comment:
I'd love to get this code abstracted into a function -- it seems like we have it in a lot of places.

But not in this PR!

It almost feels like `getProviderFromRequestOrDefault()` should return a `ProviderBuilder` (messy name for a provider with unknown aspects) rather than a `db.Provider`, which is leaking the database layer into other logic.